madmom/evaluation/chords.py

```diff
@@ -0,0 +1,897 @@
+        containing a numeric representation of chords.
+
+    """
+    crds = np.zeros(len(labels), dtype=CHORD_DTYPE)
+    cache = {}
+    for i, lbl in enumerate(labels):
+        cv = cache.get(lbl, None)
+        if cv is None:
+            cv = chord(lbl)
+            cache[lbl] = cv
+            crds[i] = cv
+        else:
+            crds[i] = cv
+
+    return crds
+
+
+def chord(label):
+    """
+    Transform a chord label into the internal numeric represenation of
+    (root, bass, intervals array).
+
+    Parameters
+    ----------
+    label : str
+        Chord label
+
+    Returns
+    -------
+    tuple
+        Numeric representation of the chord: (root, bass, intervals array)
+
+    """
+    if label == 'N':
+        return NO_CHORD
+    if label == 'X':
+        return UNKNOWN_CHORD
+
+    c_idx = label.find(':')
+    s_idx = label.find('/')
+
+    if c_idx == -1:
+        quality_str = 'maj'
+        if s_idx == -1:
+            root_str = label
+            bass_str = ''
+        else:
+            root_str = label[:s_idx]
+            bass_str = label[s_idx + 1:]
+    else:
+        root_str = label[:c_idx]
+        if s_idx == -1:
+            quality_str = label[c_idx + 1:]
+            bass_str = ''
+        else:
+            quality_str = label[c_idx + 1:s_idx]
+            bass_str = label[s_idx + 1:]
+
+    root = pitch(root_str)
+    bass = interval(bass_str) if bass_str else 0
+    ivs = chord_intervals(quality_str)
+    ivs[bass] = 1
+
+    return root, bass, ivs
+
+
+_l = [0, 1, 1, 0, 1, 1, 1]
+_chroma_id = (np.arange(len(_l) * 2) + 1) + np.array(_l + _l).cumsum() - 1
+
+
+def modify(base, modifier):
+    """
+    Modify a pitch class in integer representation by a given modifier string.
+    A modifier string can be any sequence of 'b' (one semitone down)
+    and '#' (one semitone up).
+
+    Parameters
+    ----------
+    base : int
+        Pitch class as integer
+    modifier : str
+        String of modifiers ('b' or '#')
+
+    Returns
+    -------
+    int
+        Modified root note
+
+    """
+    for m in modifier:
+        if m == 'b':
+            base -= 1
+        elif m == '#':
+            base += 1
+        else:
+            raise ValueError('Unknown modifier: {}'.format(m))
+    return base
+
+
+def pitch(pitch_str):
+    """
+    Converts a string representation of a pitch class (consisting of root
+    note and modifiers) to an integer representation.
+
+    Parameters
+    ----------
+    pitch_str : str
+        String representation of a pitch class
+
+    Returns
+    -------
+    int
+        Integer representation of a pitch class
+    """
+    return modify(_chroma_id[(ord(pitch_str[0]) - ord('C')) % 7],
+                  pitch_str[1:]) % 12
+
+
+def interval(s):
+    """
+    Converts a string representation of a musical interval into a semitone
+    integer (e.g. a minor seventh 'b7' into 10, because it is 10 semitones
+    above its base note).
```

Comment:
Same as above ('convert"). Additionally PEP 257 suggests that "multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description." However, I am not sure if it could be split like this. I don't mind if not.